### ISSUE TITLE
implement norm{} syntax for SOC constraints

### ISSUE THREAD

**Issue by mlubin** [JuliaOpt/JuMP.jl#434]
Idea:
```
@addConstraint(m, norm{ a[i]*x[i]+b[i], i=1:N } <= sum{c[i]*x[i],i=1:N})
```
Need to bikeshed the name. Also question of if we want to allow indexing over multiple dimensions (with conditions?) similar to the ``sum{}`` syntax.

**Comment by IainNZ** [comment]
That is exactly the syntax I want. Multiple dimensions could work, but I'd treat it as a vector norm, not a vector norm.
I think it could be nice to have norm1, norminf too.

**Comment by joehuchette** [comment]
+1 on the syntax. I agree with (what I think) Iain is trying to say that it should be a vector norm for multidimensional arrays. It'd be ideal to wrap the ``p=1,2,\infty`` into a kwarg, but that will clash with the way we specify index sets.

**Comment by IainNZ** [comment]
Hah yes edited. Yeah I couldn't think of a good way to do the norm type without some somewhat ambiguous syntax. Plus, don't really want to support general `p` right now

**Comment by mlubin** [comment]
I think we should focus on the 2-norm for the immediate future but design the syntax so that it's extensible. ``norm2`` or ``norm_2`` maybe?

**Comment by joehuchette** [comment]
Agreed on focusing on 2-norms but keeping on an eye on making it extensible.

**Comment by joehuchette** [comment]
``norm2``

**Comment by joehuchette** [comment]
Did a bit of work on this on [a branch](https://github.com/JuliaOpt/JuMP.jl/compare/norms!), but I didn't get very far.

**Comment by joehuchette** [comment]
Bump on this, would be nice to incorporate with my SDP branch.

**Comment by mlubin** [comment]
I'm not going to have a chance to hack on this in the next couple of weeks

**Comment by mlubin** [comment]
Closed by #465
